ipaclient/plugins/dns.py

```diff
@@ -431,11 +489,15 @@ def output_for_cli(self, textui, output, *args, **options):
         }
 
         result = output.get('result', {})
-        for key in ('ipa_records', 'location_records'):
-            if result.get(key):
-                textui.print_indented(u'{}:'.format(labels[key]), indent=1)
-                for val in sorted(result[key]):
-                    textui.print_indented(val, indent=2)
-                textui.print_line(u'')
+
+        self._standard_output(textui, result, labels)
+
+        out = options.get('out')  # output to file
+        if out:
+            try:
+                with open(out, "w") as f:
+                    self._nsupdate_output_file(f, result, labels)
+            except (OSError, IOError) as e:
+                raise errors.FileError(reason=unicode(e))
```

Comment:
This block should be moved to `forward()`, `output_for_cli()` is supposed to just format the result on the CLI, not write stuff to files.